internal/controller/api_to_config.go

```diff
@@ -255,111 +255,169 @@ func passwordForNeighbor(n v1beta1.Neighbor, passwordSecrets map[string]corev1.S
-	communitiesForPrefix := communities.communitiesForPrefixV4
-	largeCommunitiesForPrefix := communities.largeCommunitiesForPrefixV4
-	if ipFamily == ipfamily.IPv6 {
-		communitiesForPrefix = communities.communitiesForPrefixV6
-		largeCommunitiesForPrefix = communities.largeCommunitiesForPrefixV6
+func neighborHasIPFamily(neighbor *frr.NeighborConfig, ipFamily ipfamily.Family) bool {
+	if neighbor.IPFamily == ipfamily.DualStack {
+		return true
 	}
-	for p, c := range communitiesForPrefix {
-		adv, ok := advs[p]
-		if !ok {
-			return fmt.Errorf("community associated to non existing prefix %s", p)
+	return neighbor.IPFamily == ipFamily
+}
+
+func prefixesToAdvertiseForFamily(toAdvertise v1beta1.Advertise, prefixesInRouter []string, ipFamily ipfamily.Family) sets.Set[string] {
+	res := sets.New[string]()
+	prefixesForFamily := ipfamily.FilterPrefixes(prefixesInRouter, ipFamily)
+	if toAdvertise.Allowed.Mode == v1beta1.AllowAll {
+		for _, p := range prefixesForFamily {
+			res.Insert(p)
 		}
-		adv.Communities = sets.List(c)
+		return res
 	}
 
-	for p, c := range largeCommunitiesForPrefix {
-		adv, ok := advs[p]
-		if !ok {
-			return fmt.Errorf("large community associated to non existing prefix %s", p)
+	for _, p := range toAdvertise.Allowed.Prefixes {
+		prefixFamily := ipfamily.ForCIDRString(p)
+		if prefixFamily == ipFamily {
+			res.Insert(p)
 		}
-		adv.LargeCommunities = sets.List(c)
 	}
-	return nil
+	return res
 }
 
-// setLocalPrefToAdvertisements takes the given localPrefPrefixes and fills the relevant fields to the advertisements contained in the advs map.
-func setLocalPrefToAdvertisements(advs map[string]*frr.OutgoingFilter, localPrefs localPrefPrefixes, ipFamily ipfamily.Family) error {
-	localPrefsForPrefix := localPrefs.localPrefForPrefixV4
-	if ipFamily == ipfamily.IPv6 {
-		localPrefsForPrefix = localPrefs.localPrefForPrefixV6
+func frrIPFamily(ipFamily ipfamily.Family) string {
+	if ipFamily == "ipv6" {
+		return "ipv6"
 	}
+	return "ip"
+}
 
-	for p, lp := range localPrefsForPrefix {
-		adv, ok := advs[p]
-		if !ok {
-			return fmt.Errorf("localPref associated to non existing prefix %s", p)
-		}
-		adv.LocalPref = lp
+func localPrefPrefixListName(neighborID string, localPreference uint32, ipFamily string) string {
+	return fmt.Sprintf("%s-%d-%s-localpref-prefixes", neighborID, localPreference, ipFamily)
+}
+
+func communityPrefixListName(neighborID string, comm community.BGPCommunity, ipFamily string) string {
+	if community.IsLarge(comm) {
+		return fmt.Sprintf("%s-large:%s-%s-community-prefixes", neighborID, comm, ipFamily)
 	}
+	return fmt.Sprintf("%s-%s-%s-community-prefixes", neighborID, comm, ipFamily)
+}
 
-	return nil
+func communityPrefixListKey(comm community.BGPCommunity, ipFamily string) string {
+	return fmt.Sprintf("%s-%s", comm, ipFamily)
+}
+
+func localPrefPrefixListKey(localPref uint32, ipFamily string) string {
```

Comment:
can we rename ipfamily to frrIPFamily (and in the other functions we assume it is frr string)?